### DISCUSSION TITLE
Script run.sh

### DISCUSSION THREAD

**Discussion by anibalcontreras** [IIC2133-PUC/2024-1#156] [Tarea 2] [answered]
Hola buenas, tengo una duda respecto al script run.sh
Desde la tarea 0 me cree mi propio script para correr los tests y que he reutilizado para todas las tareas. Logré hacer la primera parte y tengo todos los outputs correctos.
Sin embargo con el script me sale esto
![Screenshot 2024-05-25 at 17 54 34](https://github.com/IIC2133-PUC/2024-1/assets/80776558/699b5abb-f4da-4d4a-b9e7-1659344fcdb3)
Mi pregunta es si van a usar este script para corregir la tarea o van a corregir como lo han hecho para la tarea 0 y tarea 1.
Gracias de antemano!

**Comment by benjavicente** [comment, marked as answer]
El script entregado tiene la misma lógica que el corrector automático (cambia como está implementado, pero hace lo mismo) 🤔

Falta contexto, como por ejemplo, cuál es el path de carpeta de tests, para ver que puede estar fallando.

**Reply by anibalcontreras** [reply]
Tuve un problema con los directorios, asi que ahí me funcionó perfecto!
Tengo otra duda y aprovechando que también tiene que ver con el script pregunto por acá.
Al correr el script, este lo hace en el tiempo máximo de 10 segundos por tests. Sin embargo, al correr manualmente los tests hard con valgrind para verificar los leaks de memoria estos demoran bastante mas que los 10 segundos pedidos. Para considerar que el output se dé en el tiempo correcto me guío por la ejecución del script run.sh o por el tiempo que se demora en ejecutar con valgrind
Gracias!!

**Reply by benjavicente** [reply]
Solo se ve los temas de memoria con valgrind. El tiempo se ve sin.

Valgrind ha llegado a demorarse x100 más en ciertos casos, así que lo ejecutamos aparte 😅
